### PR TITLE
Update Arcade and other tool packages

### PR DESCRIPTION
**Pull request by crummel**

### CODE REVIEW

**Comment by crummel** [comment]
A couple small prebuilt -> tarball things to fix yet, but this should be reviewable.

File: build-source-tarball.sh [hunk 1]
```diff
@@ -233,6 +233,9 @@ done
 echo 'Copying source-built packages to tarball to replace packages needed before they are built...'
 mkdir -p $TARBALL_ROOT/packages/source-built
 cp -r $SCRIPT_ROOT/Tools/source-built/coreclr-tools $TARBALL_ROOT/packages/source-built/
+cp $SCRIPT_ROOT/bin/obj/x64/Release/blob-feed/packages/*Arcade*.nupkg $TARBALL_ROOT/packages/source-built/
+cp $SCRIPT_ROOT/bin/obj/x64/Release/blob-feed/packages/*SourceLink*.nupkg $TARBALL_ROOT/packages/source-built/
+cp $SCRIPT_ROOT/bin/obj/x64/Release/blob-feed/packages/*Build*Tasks*.nupkg $TARBALL_ROOT/packages/source-built/
```
Comment (dseefeld) [inline build-source-tarball.sh]: These should come from the source-built artifacts tarball.  If we need a version update, maybe this needs my change to update these to 3.0.100 version of that?  My PR is failing for that on building arcade in the tarball.  I'm wondering if I need these Arcade updates?

File: patches/core-setup/0004-core-setup-9f64d335ef219be259997eea043e9c5b572f3959.patch [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+diff --git a/src/test/Assets/TestProjects/Directory.Build.targets b/src/test/Assets/TestProjects/Directory.Build.targets
+index 103be55d..c8e33d3a 100644
+--- a/src/test/Assets/TestProjects/Directory.Build.targets
++++ b/src/test/Assets/TestProjects/Directory.Build.targets
+@@ -6,7 +6,7 @@
+     versions. Remove them before the SDK tries to download them.
+   -->
+   <Target Name="RemoveUpstackKnownFrameworkReferences"
+-          BeforeTargets="ResolveFrameworkReferences">
++          BeforeTargets="ProcessFrameworkReferences">
```
Comment (dseefeld) [inline patches/core-setup/0004-core-setup-9f64d335ef219be259997eea043e9c5b572f3959.patch]: I don't get what's happening here.

File: build-source-tarball.sh (hunk 1 above)
Comment (crummel) [reply build-source-tarball.sh]: I updated the source-built artifacts tarball but was still running into issues with the version numbers here.  Logged https://github.com/dotnet/source-build/issues/1321 to fix this.

File: patches/core-setup/0004-core-setup-9f64d335ef219be259997eea043e9c5b572f3959.patch (hunk 2 above)
Comment (crummel) [reply patches/core-setup/0004-core-setup-9f64d335ef219be259997eea043e9c5b572f3959.patch]: This was an obsolete patch; removed.

**Comment by crummel** [comment]
Merging with known timeout issue.

**Pull request merged**